type-check.ts

```diff
@@ -261,6 +272,19 @@ export function tcStmt(env: GlobalTypeEnv, locals: LocalTypeEnv, stmt: Stmt<Anno
         throw new TypeCheckError(SRC, `field \`${stmt.field}\` expected type: ${JSON.stringify(fields.get(stmt.field).tag)}, got value of type ${JSON.stringify(tVal.a.type.tag)}`,
           tVal.a);
       return { ...stmt, a: { ...stmt.a, type: NONE }, obj: tObj, value: tVal };
+    case "index-assign":
```

Comment:
From set/dict group,
we'll have a huge conflict here (due to "[]" method)
A simple question, Why your annotation will have "a.type.tag" instead of "a.tag"?

Reply:
Hi,
When we did our last merge, we incorporated code from [this PR](https://github.com/ucsd-cse231-s22/chocopy-wasm-compiler-A/pull/16) which was already merged at that time.

Reply:
No worry, we implement our "index" and "index-assign" first